plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -62,6 +64,7 @@
     // OPTIMIZE only. Coordinator-only
     private final boolean recordScannedFiles;
     private final Optional<DataSize> maxScannedFileSize;
+    private final Set<ColumnHandle> constraintColumns;
```

Comment:
constraintColumns comes from `Sets.union(table.getConstraintColumns(), constraint.getPredicateColumns().orElseGet(ImmutableSet::of))`, it is indeed derived from `constraint`, here is the same as the hive connector. But it cannot be obtained from `TupleDomain` I think, because this is a calculated column, such as the partition calculation that udf participates in. What exactly are you referring to?

Reply:
`constraintColumns` is derived from `Sets.union(table.getConstraintColumns(), constraint.getPredicateColumns().orElseGet(ImmutableSet::of))`, which indeed derived from the `constraint` and is in line with the implementation found in the Hive connector.